rules/integrations/aws/credential_access_retrieve_secure_string_parameters_via_ssm.toml

```diff
@@ -0,0 +1,118 @@
+- **Inspect Related CloudTrail Events**: Look for other related events in CloudTrail to see if there was unusual activity before or after this event, such as unusual login attempts, changes to permissions, or other API calls that could indicate broader unauthorized actions.
+
+### False Positive Analysis
+
+- **Legitimate Administrative Use**: Verify if the decryption of SecureString parameters is a common practice for the user’s role, particularly if used in automation scripts or deployment processes like those involving Terraform or similar tools.
+
+### Response and Remediation
+
+- **Immediate Verification**: Contact the user or team responsible for the API call to verify their intent and authorization.
+- **Review and Revise Permissions**: If the access was unauthorized, review the permissions assigned to the user or role to ensure they align with the principle of least privilege.
+- **Audit Parameter Access Policies**: Ensure that policies governing access to SecureString parameters are strict and audit logs are enabled to track access with decryption.
+- **Incident Response**: If suspicious activity is confirmed, follow through with your organization's incident response plan to mitigate any potential security issues.
+- **Enhanced Monitoring and Alerting**: Strengthen monitoring rules to detect unusual accesses to SecureString parameters, especially those that involve decryption.
+
+### Additional Information
+
+This rule focuses solely on SecureStrings in AWS Systems Manager (SSM) parameters. SecureStrings are encrypted using an AWS Key Management Service (KMS) key. When a user accesses a SecureString parameter, they can specify whether the parameter should be decrypted. If the user specifies that the parameter should be decrypted, the decrypted value is returned in the response.
+"""
+references = [
+    "https://docs.aws.amazon.com/vsts/latest/userguide/systemsmanager-getparameter.html",
+    "https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html",
+]
+risk_score = 47
+rule_id = "fd332492-0bc6-11ef-b5be-f661ea17fbcc"
+setup = "This rule requires that AWS CloudTrail logs are ingested into the Elastic Stack. Ensure that the AWS integration is properly configured to collect AWS CloudTrail logs. This rule also requires event logging for AWS Systems Manager (SSM) API actions which can be enabled in CloudTrail's data events settings.\n"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: AWS Systems Manager",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset: "aws.cloudtrail"
+    and event.provider: "ssm.amazonaws.com" and event.action: (GetParameters or GetParameter) and event.outcome: success
+    and aws.cloudtrail.request_parameters: *withDecryption=true*
```

Comment:
If we have a field that shows parameter.type: SecureString I would suggest adding that here. Requests for Strings and Stringlist types ignore this withDecrytion boolean however I'm not sure they would return a failed response. So we may want to avoid false positives by making sure to only alert on requests for SecureString type.

Reply:
This is a great question Isai!

The documentation for this API call states that the response should include a parameter type as you explained, however, from the data I was unable to locate the typical `response_elements` field where this data would exist. I have included a data sample below. 

I'm going to put this back into `in-design` status so I can dig a bit deeper. I'll attempt to replicate the API call and capture the response. If the response does indeed include this information, we should create an integration issue request to have it investigated and hopefully fixed.

```
{
  "_index": ".ds-logs-aws.cloudtrail-default-2024.04.09-000001",
  "_id": "a92434ea09-000000529436",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "ip-172-31-95-103",
      "id": "f14d530d-b7f2-4dbd-b122-28582c2a767c",
      "ephemeral_id": "69b4fa20-756a-4d41-8325-7613b13a01b2",
      "type": "filebeat",
      "version": "8.13.2"
    },
    "log": {
      "file": {
        "path": "https://asperitas-security-logs.s3.us-east-1.amazonaws.com/AWSLogs/x/CloudTrail/us-east-1/2024/04/12/x_CloudTrail_us-east-1_20240412T1615Z_CINQrdyxpT8CLclQ.json.gz"
      },
      "offset": 529436
    },
    "elastic_agent": {
      "id": "f14d530d-b7f2-4dbd-b122-28582c2a767c",
      "version": "8.13.2",
      "snapshot": false
    },
    "source": {
      "geo": {
        "region_iso_code": "US-OH",
        "continent_name": "North America",
        "city_name": "Massillon",
        "country_iso_code": "US",
        "country_name": "United States",
        "region_name": "Ohio",
        "location": {
          "lon": x,
          "lat": x
        }
      },
      "as": {
        "number": 12097,
        "organization": {
          "name": "MASSCOM"
        }
      },
      "address": "x",
      "ip": "x"
    },
    "tags": [
      "preserve_original_event",
      "forwarded",
      "aws-cloudtrail"
    ],
    "cloud": {
      "region": "us-east-1",
      "account": {
        "id": "x"
      }
    },
    "input": {
      "type": "aws-s3"
    },
    "@timestamp": "2024-04-12T16:14:46.000Z",
    "ecs": {
      "version": "8.0.0"
    },
    "related": {
      "user": [
        "stratus"
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "aws.cloudtrail"
    },
    "tls": {
      "cipher": "ECDHE-RSA-AES128-GCM-SHA256",
      "client": {
        "server_name": "ssm.us-east-1.amazonaws.com"
      },
      "version": "1.2",
      "version_protocol": "tls"
    },
    "aws": {
      "s3": {
        "bucket": {
          "name": "asperitas-security-logs",
          "arn": "arn:aws:s3:::asperitas-security-logs"
        },
        "object": {
          "key": "AWSLogs/x/CloudTrail/us-east-1/2024/04/12/x_CloudTrail_us-east-1_20240412T1615Z_CINQrdyxpT8CLclQ.json.gz"
        }
      },
      "cloudtrail": {
        "event_version": "1.08",
        "flattened": {
          "request_parameters": {
            "withDecryption": true,
            "name": "/credentials/stratus-red-team/credentials-2"
          }
        },
        "event_type": "AwsApiCall",
        "read_only": true,
        "user_identity": {
          "access_key_id": "AKIA47CRWDCFXZ3V7UXR",
          "type": "IAMUser",
          "arn": "arn:aws:iam::x:user/stratus"
        },
        "recipient_account_id": "x",
        "event_category": "Management",
        "request_parameters": "{withDecryption=true, name=/credentials/stratus-red-team/credentials-2}",
        "request_id": "27fd6c1f-74df-45bc-899a-035968f5f1f3",
        "management_event": true
      }
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2024-04-12T16:18:23Z",
      "original": "{\"eventVersion\":\"1.08\",\"userIdentity\":{\"type\":\"IAMUser\",\"principalId\":\"AIDA47CRWDCFTQGUB5FBF\",\"arn\":\"arn:aws:iam::x:user/stratus\",\"accountId\":\"x\",\"accessKeyId\":\"AKIA47CRWDCFXZ3V7UXR\",\"userName\":\"stratus\"},\"eventTime\":\"2024-04-12T16:14:46Z\",\"eventSource\":\"ssm.amazonaws.com\",\"eventName\":\"GetParameter\",\"awsRegion\":\"us-east-1\",\"sourceIPAddress\":\"x\",\"userAgent\":\"APN/1.0 HashiCorp/1.0 Terraform/1.1.2 (+https://www.terraform.io) terraform-provider-aws/3.76.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.157 (go1.19.3; darwin; arm64) HashiCorp-terraform-exec/0.17.3\",\"requestParameters\":{\"name\":\"/credentials/stratus-red-team/credentials-2\",\"withDecryption\":true},\"responseElements\":null,\"requestID\":\"27fd6c1f-74df-45bc-899a-035968f5f1f3\",\"eventID\":\"76d32334-717b-4b99-8f09-d6dd2c389833\",\"readOnly\":true,\"resources\":[{\"accountId\":\"x\",\"ARN\":\"arn:aws:ssm:us-east-1:x:parameter/credentials/stratus-red-team/credentials-2\"}],\"eventType\":\"AwsApiCall\",\"managementEvent\":true,\"recipientAccountId\":\"x\",\"eventCategory\":\"Management\",\"tlsDetails\":{\"tlsVersion\":\"TLSv1.2\",\"cipherSuite\":\"ECDHE-RSA-AES128-GCM-SHA256\",\"clientProvidedHostHeader\":\"ssm.us-east-1.amazonaws.com\"}}",
      "provider": "ssm.amazonaws.com",
      "created": "2024-04-12T16:18:12.723Z",
      "kind": "event",
      "action": "GetParameter",
      "id": "76d32334-717b-4b99-8f09-d6dd2c389833",
      "type": [
        "info"
      ],
      "dataset": "aws.cloudtrail",
      "outcome": "success"
    },
    "user": {
      "name": "stratus",
      "id": "AIDA47CRWDCFTQGUB5FBF"
    },
    "user_agent": {
      "original": "APN/1.0 HashiCorp/1.0 Terraform/1.1.2 (+https://www.terraform.io) terraform-provider-aws/3.76.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.157 (go1.19.3; darwin; arm64) HashiCorp-terraform-exec/0.17.3",
      "name": "aws-sdk-go",
      "device": {
        "name": "Other"
      },
      "version": "1.44.157"
    }
  },
  "fields": {
    "aws.cloudtrail.request_parameters.text": [
      "{withDecryption=true, name=/credentials/stratus-red-team/credentials-2}"
    ],
    "elastic_agent.version": [
      "8.13.2"
    ],
    "tls.version_protocol": [
      "tls"
    ],
    "user_agent.original.text": [
      "APN/1.0 HashiCorp/1.0 Terraform/1.1.2 (+https://www.terraform.io) terraform-provider-aws/3.76.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.157 (go1.19.3; darwin; arm64) HashiCorp-terraform-exec/0.17.3"
    ],
    "agent.name.text": [
      "ip-172-31-95-103"
    ],
    "source.geo.region_name": [
      "Ohio"
    ],
    "source.ip": [
      "x"
    ],
    "agent.name": [
      "ip-172-31-95-103"
    ],
    "user_agent.version": [
      "1.44.157"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "source.geo.region_iso_code": [
      "US-OH"
    ],
    "aws.cloudtrail.management_event": [
      "true"
    ],
    "event.kind": [
      "event"
    ],
    "aws.cloudtrail.user_identity.arn": [
      "arn:aws:iam::x:user/stratus"
    ],
    "event.outcome": [
      "success"
    ],
    "source.geo.city_name": [
      "Massillon"
    ],
    "tls.version": [
      "1.2"
    ],
    "user_agent.original": [
      "APN/1.0 HashiCorp/1.0 Terraform/1.1.2 (+https://www.terraform.io) terraform-provider-aws/3.76.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.157 (go1.19.3; darwin; arm64) HashiCorp-terraform-exec/0.17.3"
    ],
    "event.original": [
      "{\"eventVersion\":\"1.08\",\"userIdentity\":{\"type\":\"IAMUser\",\"principalId\":\"AIDA47CRWDCFTQGUB5FBF\",\"arn\":\"arn:aws:iam::x:user/stratus\",\"accountId\":\"x\",\"accessKeyId\":\"AKIA47CRWDCFXZ3V7UXR\",\"userName\":\"stratus\"},\"eventTime\":\"2024-04-12T16:14:46Z\",\"eventSource\":\"ssm.amazonaws.com\",\"eventName\":\"GetParameter\",\"awsRegion\":\"us-east-1\",\"sourceIPAddress\":\"x\",\"userAgent\":\"APN/1.0 HashiCorp/1.0 Terraform/1.1.2 (+https://www.terraform.io) terraform-provider-aws/3.76.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.157 (go1.19.3; darwin; arm64) HashiCorp-terraform-exec/0.17.3\",\"requestParameters\":{\"name\":\"/credentials/stratus-red-team/credentials-2\",\"withDecryption\":true},\"responseElements\":null,\"requestID\":\"27fd6c1f-74df-45bc-899a-035968f5f1f3\",\"eventID\":\"76d32334-717b-4b99-8f09-d6dd2c389833\",\"readOnly\":true,\"resources\":[{\"accountId\":\"x\",\"ARN\":\"arn:aws:ssm:us-east-1:x:parameter/credentials/stratus-red-team/credentials-2\"}],\"eventType\":\"AwsApiCall\",\"managementEvent\":true,\"recipientAccountId\":\"x\",\"eventCategory\":\"Management\",\"tlsDetails\":{\"tlsVersion\":\"TLSv1.2\",\"cipherSuite\":\"ECDHE-RSA-AES128-GCM-SHA256\",\"clientProvidedHostHeader\":\"ssm.us-east-1.amazonaws.com\"}}"
    ],
    "cloud.region": [
      "us-east-1"
    ],
    "user.id": [
      "AIDA47CRWDCFTQGUB5FBF"
    ],
    "input.type": [
      "aws-s3"
    ],
    "log.offset": [
      529436
    ],
    "user_agent.name": [
      "aws-sdk-go"
    ],
    "data_stream.type": [
      "logs"
    ],
    "related.user": [
      "stratus"
    ],
    "tags": [
      "preserve_original_event",
      "forwarded",
      "aws-cloudtrail"
    ],
    "event.provider": [
      "ssm.amazonaws.com"
    ],
    "agent.id": [
      "f14d530d-b7f2-4dbd-b122-28582c2a767c"
    ],
    "ecs.version": [
      "8.0.0"
    ],
    "event.created": [
      "2024-04-12T16:18:12.723Z"
    ],
    "aws.cloudtrail.event_version": [
      "1.08"
    ],
    "agent.version": [
      "8.13.2"
    ],
    "source.as.number": [
      12097
    ],
    "aws.cloudtrail.read_only": [
      true
    ],
    "aws.cloudtrail.event_category": [
      "Management"
    ],
    "aws.cloudtrail.user_identity.type": [
      "IAMUser"
    ],
    "aws.s3.bucket.arn": [
      "arn:aws:s3:::asperitas-security-logs"
    ],
    "aws.cloudtrail.recipient_account_id": [
      "x"
    ],
    "aws.cloudtrail.request_id": [
      "27fd6c1f-74df-45bc-899a-035968f5f1f3"
    ],
    "tls.cipher": [
      "ECDHE-RSA-AES128-GCM-SHA256"
    ],
    "user.name": [
      "stratus"
    ],
    "source.geo.location": [
      {
        "coordinates": [
          -81.4971,
          40.8133
        ],
        "type": "Point"
      }
    ],
    "source.address": [
      "x"
    ],
    "aws.cloudtrail.flattened.request_parameters": [
      {
        "withDecryption": true,
        "name": "/credentials/stratus-red-team/credentials-2"
      }
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "aws"
    ],
    "source.geo.country_iso_code": [
      "US"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "aws.cloudtrail.event_type": [
      "AwsApiCall"
    ],
    "aws.s3.bucket.name": [
      "asperitas-security-logs"
    ],
    "source.as.organization.name.text": [
      "MASSCOM"
    ],
    "elastic_agent.id": [
      "f14d530d-b7f2-4dbd-b122-28582c2a767c"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "source.as.organization.name": [
      "MASSCOM"
    ],
    "source.geo.continent_name": [
      "North America"
    ],
    "tls.client.server_name": [
      "ssm.us-east-1.amazonaws.com"
    ],
    "event.action": [
      "GetParameter"
    ],
    "event.ingested": [
      "2024-04-12T16:18:23.000Z"
    ],
    "@timestamp": [
      "2024-04-12T16:14:46.000Z"
    ],
    "cloud.account.id": [
      "x"
    ],
    "aws.cloudtrail.user_identity.access_key_id": [
      "AKIA47CRWDCFXZ3V7UXR"
    ],
    "data_stream.dataset": [
      "aws.cloudtrail"
    ],
    "event.type": [
      "info"
    ],
    "log.file.path": [
      "https://asperitas-security-logs.s3.us-east-1.amazonaws.com/AWSLogs/x/CloudTrail/us-east-1/2024/04/12/x_CloudTrail_us-east-1_20240412T1615Z_CINQrdyxpT8CLclQ.json.gz"
    ],
    "agent.ephemeral_id": [
      "69b4fa20-756a-4d41-8325-7613b13a01b2"
    ],
    "aws.cloudtrail.request_parameters": [
      "{withDecryption=true, name=/credentials/stratus-red-team/credentials-2}"
    ],
    "event.id": [
      "76d32334-717b-4b99-8f09-d6dd2c389833"
    ],
    "source.geo.country_name": [
      "United States"
    ],
    "user_agent.device.name": [
      "Other"
    ],
    "aws.s3.object.key": [
      "AWSLogs/x/CloudTrail/us-east-1/2024/04/12/x_CloudTrail_us-east-1_20240412T1615Z_CINQrdyxpT8CLclQ.json.gz"
    ],
    "event.dataset": [
      "aws.cloudtrail"
    ],
    "user.name.text": [
      "stratus"
    ]
  }
}
````

Reply:
Digging into this a bit, it looks like the response_elements in the cloudtrail log are actually NULL, which is why they are not available in ES.

<img width="781" alt="Screenshot 2024-06-03 at 12 55 15 PM" src="https://github.com/elastic/detection-rules/assets/99630311/07df662a-f07e-4384-bfb1-6da2c81a87aa">

Reply:
Ah, it is documented that only for create, delete or change events are a response included. Since this API call is a `Get` response element is null.

https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-event-reference-record-contents.html

I am curious if it can be invoked then through the request, however, even if so we cannot rely on the customer always requesting the response during the API call.

Reply:
Note that there is no configuration setting that can be used to always capture API responses via CloudTrail. Capturing responses for this would be something required outside of CloudTrail such as setting up AWS Lambda function, or enabling and configuring CloudWatch specifically. Therefore we cannot recommend the user to do so in the rule nor will this data generate with the response.

Reply:
Since `String` and `Stringlist` ignore that flag they likely won't even send it as part of the `request.parameter`, I hadn't thought of that...thanks for digging.

Reply:
Thanks for bringing this up or else I would have not determined why response elements could be NULL!